### ISSUE TITLE
DiscPage is not compatible with the "Checklist" plugin

### ISSUE THREAD

**Issue by sylque** [sylque/discpage#19]
When the [Checklist](https://meta.discourse.org/t/checklist-plugin-interactive-checkboxes-made-simple/36362) plugin is installed, static pages render like normal Discourse topics (instead of static pages) and the console shows this error:
```
Uncaught (in promise) TypeError: Cannot read property 'can_edit' of undefined
    at checklistSyntax (javascripts/discourse/initializers/checklist:29)
    at eval (discourse/app/lib/plugin-api:182)
    at eval (discourse/app/widgets/post-cooked:89)
    at Array.forEach (<anonymous>)
    at PostCooked._decorateAndAdopt (discourse/app/widgets/post-cooked:88)
    at PostCooked.init (discourse/app/widgets/post-cooked:79)
    at N (javascripts/discourse/lib/lib:253)
    at S (javascripts/discourse/lib/lib:484)
    at eval (javascripts/discourse/lib/lib:923)
```

**Comment by sylque** [comment]
This is due to a bug in DiscPage. Because of the current method DiscPage uses to render a post, any plugin that uses the post model to  alter the rendering of a post will fail. 

In a final solution, DiscPage should use a cleaner method to render a post. In more technical terms, I need to find a way to get an up-to-date decoratorHelper out of the context of a post route.

Because this looks hard to do, I will first try to implement a quick workaround.

**Comment by sylque** [comment]
Temporary fix in 1.0.38
